app/javascript/packs/user_profile/body.jsx

```diff
@@ -59,15 +65,91 @@ class Body extends React.Component {
   }
 
 
+  handleUpdate(e) {
+    e.preventDefault();
+    this.setState({ editable: !this.state.editable });
+    const first_name = this.state.first_name;
+    const last_name = this.state.last_name;
+    $.ajax({
+      url: `/api/v1/profile`,
+      type: 'PATCH',
+      beforeSend(xhr) { xhr.setRequestHeader('X-CSRF-Token', $('meta[name="csrf-token"]').attr('content')); },
+      data: { first_name, last_name },
+      success: () => {
+        this.msg.success('Successfully updated');
+        this.loadUser();
+      },
+      error: (xhr) => {
+        this.msg.error($.parseJSON(xhr.responseText).errors);
+      },
+    });
+  }
+
+  handleEdit() {
+    if (this.state.editable) {
+      const user = { first_name: this.state.user.first_name, last_name: this.state.user.last_name };
+      this.props.handleUpdate(user);
+    }
+    this.setState({ editable: !this.state.editable });
+  }
+
+  handleBack() {
+    this.setState({ editable: !this.state.editable });
+  }
+
+  handleNameChange(event) {
+    this.setState({ first_name: event.target.value });
+  }
+
+  handleLastNameChange(event) {
+    this.setState({ last_name: event.target.value });
+  }
+
+
   render() {
     const avatar = <img src={this.state.user.photo} alt="avatar" className="img-responsive" />;
+    const first_name = this.state.user.first_name;
```

Comment:
in js it is better to use js syxtax styleguides.
https://github.com/airbnb/javascript
`firstName`
Please review your code and update to the style guides.
Also, it would be nice to update tests.

Reply:
fixed + comitted